network-store-iidm-impl/src/main/java/com/powsybl/network/store/iidm/impl/TerminalImpl.java

```diff
@@ -230,7 +225,7 @@ boolean getConnectingSwitches(Predicate<Switch> isSwitchOperable, Set<SwitchImpl
                         sw.ifPresent(switchForConnection::add);
                     }
                 });
-            done = true;
+            done = !switchForConnection.isEmpty();
```

Comment:
Are you sure about that ? That's not how it is implemented in powsybl-core

Reply:
Without that modification, the tck test nodeBreakerTest in AbstractConnectionNotificationTest fail at line 124 : 

assertFalse(l1.getTerminal1().connect());

because, no modification should be done on a switch, see the assertions that just after : 

assertEquals(List.of(
                        new Event("LINE_S2S3", "beginConnect", "InitialState", true, null),
                        new Event("LINE_S2S3", "endConnect", "InitialState", null, true)),
                eventsRecorder.getEvents());

And in connectAllTerminals method of ConnectDisconnectUtil.java, the return value should be false, in case of the terminal cannot be connected ... 

if (terminal instanceof TerminalImpl<?> terminalImpl && terminalImpl.isNodeBeakerTopologyKind()
    && !terminalImpl.getConnectingSwitches(isTypeSwitchToOperate, switchesForConnection)) {
        // Exit if the terminal cannot be connected
        return false;
}

Reply:
Actually in powsybl-core this method stays like that but in the connect/disconnect methods there is an early check to return if there is no action to perform.
```
@Override
    public boolean disconnect(TerminalExt terminal, Predicate<? super SwitchImpl> isSwitchOpenable) {
        // Check the topology kind
        checkTopologyKind(terminal);

        // already disconnected?
        if (!terminal.isConnected()) {
            return false;
        }
```
Shouldn't we keep the same logic to remain consistent with in-memory implementation ?

Reply:
The implementation code is a little bit different between powsybl-core and powsybl-ntwork-store, for the terminal connect/disconnect actions ...

When the early check is done in powsybl-core, the 'beginConnect' update notification has been done (see AbstractTerminal connect method) and the 'endConnect' update notification will be done after, even if no switch modification is done ...
And we must do the same notifications in TerminalImpl connect method in network-store ...

Reply:
humm same for us, no ? we call connect -> notify "beginConnect" -> call connectNodeBreaker -> notify "endConnect". I feel like if we do the early check in connectNodeBreaker it's the same behavior as in powsybl-core, no ?